src/connectionhealthpolicy/SendingAudioFailureConnectionHealthPolicy.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Logger from '../logger/Logger';
+import BaseConnectionHealthPolicy from './BaseConnectionHealthPolicy';
+import ConnectionHealthData from './ConnectionHealthData';
+import ConnectionHealthPolicy from './ConnectionHealthPolicy';
+import ConnectionHealthPolicyConfiguration from './ConnectionHealthPolicyConfiguration';
+
+export default class SendingAudioFailureConnectionHealthPolicy
+  extends BaseConnectionHealthPolicy
+  implements ConnectionHealthPolicy {
+  private readonly secondsToConsider: number;
+  private readonly initialWaitTimeMs: number;
+  private readonly coolDownTimeMs: number;
+  private readonly maximumTimesToWarn: number;
+  private warnCount: number;
+  private lastWarnTimestampMs: number;
+
+  constructor(
+    private logger: Logger,
+    configuration: ConnectionHealthPolicyConfiguration,
+    data: ConnectionHealthData
+  ) {
+    super(configuration, data, 'Sending Audio Health');
+    this.secondsToConsider = configuration.sendingAudioFailureSamplesToConsider;
+    this.initialWaitTimeMs = configuration.sendingAudioFailureInitialWaitTimeMs;
+    this.maximumTimesToWarn = configuration.maximumTimesToWarn;
+    this.coolDownTimeMs = configuration.cooldownTimeMs;
+    this.lastWarnTimestampMs = 0;
+    this.warnCount = 0;
+  }
+
+  private isSendingAudioUnHealthy(): boolean {
+    const hasEnoughTimeElapsedToEvaluateStatus = !this.currentData.isConnectionStartRecent(
+      this.initialWaitTimeMs
+    );
+    const areAudioPacketsNotBeingSent =
+      this.currentData.consecutiveStatsWithNoAudioPacketsSent >= this.secondsToConsider;
+    return hasEnoughTimeElapsedToEvaluateStatus && areAudioPacketsNotBeingSent;
+  }
+
+  health(): number {
+    const didWarnRecently = Date.now() - this.lastWarnTimestampMs < this.coolDownTimeMs;
```

Comment:
Probably we should return if `didWarnRecently` is `true`, cause, then no need to calculate `isSendingAudioUnhealthy` and we can save some computations.

Reply:
`didWarnRecently` is only used when we're in the unhealthy path (i.e. returning minimumHealth) because I wanted to make sure that we notify asap when we recover, so we need to calculate the `isSendingAudioUnhealthy` each time.

Reply:
So better to move `const didWarnRecently = Date.now() - this.lastWarnTimestampMs < this.coolDownTimeMs;` inside the `this.isSendingAudioUnHealthy()` check then?